Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-FIPFSScanEngineVersionState.ps1

```diff
@@ -0,0 +1,148 @@
+                            FileBuildPart   = $getItem.VersionInfo.FileBuildPart
+                            FilePrivatePart = $getItem.VersionInfo.FilePrivatePart
+                        })
+                })
+        }
+        function TestPipeline2Version {
+            param (
+                [string]$ComputerName,
+                [scriptblock]$CatchActionFunction
+            )
+
+            Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+
+            $pipeline2FileInfo = Invoke-ScriptBlockHandler -ComputerName $ComputerName `
+                -ScriptBlock ${Function:GetItemFromExchangeInstallPath} `
+                -ArgumentList ("FIP-FS\Bin\pipeline2.dll") `
+                -CatchActionFunction $CatchActionFunction
+
+            if ($null -ne $pipeline2FileInfo) {
+                Write-Verbose "Testing pipeline2.dll version: $($pipeline2FileInfo.VersionInfo.ProductVersion)"
+
+                $pipeline2ReturnObject = ([PSCustomObject]@{
+                        isE15    = $false
+                        Affected = $false
+                    })
+
+                try {
+                    [int]$fileMajor = $pipeline2FileInfo.VersionInfo.FileMajorPart
+                    [int]$fileMinor = $pipeline2FileInfo.VersionInfo.FileMinorPart
+                    [int]$fileBuild = $pipeline2FileInfo.VersionInfo.FileBuildPart
+                    [int]$filePrivate = $pipeline2FileInfo.VersionInfo.FilePrivatePart
+
+                    if ($fileMajor -eq 15) {
+
+                        switch ($fileMinor) {
+                            0 { $pipeline2ReturnObject.isE15 = $true }
+                            1 { $pipeline2ReturnObject.Affected = (($fileBuild -le 2375) -and ($filePrivate -le 17)) }
+                            2 { $pipeline2ReturnObject.Affected = (($fileBuild -le 986) -and ($filePrivate -le 14)) }
+                            Default { Write-Verbose "Unexpected minor passed to the switch statement" }
+                        }
+                    } else {
+                        Write-Verbose "Exchange server version is not affected by the FIP-FS update issue"
+                    }
+                } catch {
+                    Write-Verbose "Error occured while detecting pipeline2.dll version"
+                    & $CatchActionFunction
+                }
+            }
+            return $pipeline2ReturnObject
+        }
+
+        function GetHighestScanEngineVersionNumber {
+            param (
+                [string]$ComputerName,
+                [scriptblock]$CatchActionFunction
+            )
+
+            Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+
+            try {
+                $scanEngineVersions = Invoke-ScriptBlockHandler -ComputerName $ComputerName `
+                    -ScriptBlock { Get-ChildItem -Path (Join-Path $env:ExchangeInstallPath "FIP-FS\Data\Engines\amd64\Microsoft\Bin") -Attributes Directory } `
```

Comment:
Should be doing the same thing here with the path and get it from the registry

Reply:
Yep, fixed.